src/rules/tryceratops/rules/verbose_raise.rs

```diff
@@ -13,28 +15,47 @@ define_violation!(
 impl Violation for VerboseRaise {
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Use 'raise' without specifying exception name")
+        format!("Use `raise` without specifying exception name")
+    }
+}
+
+#[derive(Default)]
+struct RaiseStatementVisitor<'a> {
+    raises: Vec<Option<&'a Expr>>,
+}
+
+impl<'a, 'b> Visitor<'b> for RaiseStatementVisitor<'a>
+where
+    'b: 'a,
+{
+    fn visit_stmt(&mut self, stmt: &'b Stmt) {
+        match &stmt.node {
+            StmtKind::Raise { exc, .. } => self.raises.push(exc.as_ref().map(|expr| &**expr)),
+            _ => visitor::walk_stmt(self, stmt),
```

Comment:
This will end up raising twice for this case:

```
try:
    1/0
except ZeroDivisionError as e:
    try:
        foo
    except Exception as e:
        raise e
```

But it looks like the original plugin does that too, so, oh well.

Reply:
It'll also raise here, but, that's fine:

```py
try:
    1/0
except ZeroDivisionError as e:
    e = 1
    raise e
```

Reply:
yes, i didn't like the fact that the second one raises, but the original didn't handle shadowing of the name, so i guess that is ok.

is there anything where something like that is handled in ruff?

Reply:
It's possible for us to handle these cases correctly by making this rule run on `Raise` statements rather than over `&[Excepthandler]`... We could see if the raise references a name, then do `checker.find_binding(name)`, and check if that binding comes from an except handler. It's a bit more involved, but it'd be more robust and probably a bit faster. If you're interested in trying that, cool! If not, we can still merge with this structure.

Reply:
Yeah that's fine too. We can merge as-is, especially since (AFAICT) it has parity with the existing plugin.